cwl/src/main/scala/cwl/InputParameter.scala

```diff
@@ -122,6 +123,19 @@ object InputParameter {
             } yield updated
 
           case WomArray(_, values) => values.toList.traverse(populateFiles).map(WomArray(_))
+          case WomOptionalValue(_, Some(innerValue)) => populateFiles(innerValue).map(WomOptionalValue(_))
+          case obj: WomObjectLike =>
+            // Map the values
+            obj.values.toList.traverse[ErrorOr, (String, WomValue)]({
+              case (key, value) => populateFiles(value).map(key -> _)
+            })
+              .map(_.toMap)
+              // transform to Either so we can flatMap
+              .toEither
+              // Validate new types are still valid w.r.t the object
+              .flatMap(WomObject.withTypeChecked(_, obj.womObjectTypeLike))
+              // re-transform to ErrorOr
+              .toValidated
```

Comment:
I don't think the method `withTypeChecked` is required.

```scala
// transform to Either so we can flatMap
.toEither
// Validate new types are still valid w.r.t the object
.flatMap(WomObject.withTypeChecked(_, obj.womObjectTypeLike))
// re-transform to ErrorOr
.toValidated
```

should be the same as

```scala
.flatMap(WomObject.withTypeErrorOr(_, obj.womObjectTypeLike))
```

Reply:
flatMapping `ErrorOr` requires the special `ShortCircuitingFlatMap` import whereas `Either` can naturally flatMap since 2.12 so I prefer using that even though it's a bit more verbose